web-stream-tools.d.ts

```diff
@@ -2,25 +2,39 @@
 // Contributors:
 // - Flowcrypt a.s. <human@flowcrypt.com>
 
-declare module "@openpgp/web-stream-tools" {
-    type Data = Uint8Array | string;
-
-    interface BaseStream<T extends Data> extends AsyncIterable<T> { }
+export namespace ReadableStreamInternals {
+  // copied+simplified version of ReadableStream from lib.dom.d.ts, needed to enforce type checks in WebStream below
+  interface ReadableStreamDefaultReadDoneResult {
+    done: true;
+    value?: undefined;
+  }
+  interface ReadableStreamDefaultReadValueResult<T> {
+    done: false;
+    value: T;
+  }
+  interface ReadableStreamDefaultReader<R = any> {
+    readonly closed: Promise<undefined>;
+    cancel(reason?: any): Promise<void>;
+    read(): Promise<ReadableStreamDefaultReadValueResult<R> | ReadableStreamDefaultReadDoneResult>;
+    releaseLock(): void;
+  }
+}
 
-    interface WebStream<T extends Data> extends BaseStream<T> { // copied+simplified version of ReadableStream from lib.dom.d.ts
-      readonly locked: boolean; getReader: Function; pipeThrough: Function; pipeTo: Function; tee: Function;
-      cancel(reason?: any): Promise<void>;
-    }
+type Data = Uint8Array | string;
 
-    interface NodeStream<T extends Data> extends BaseStream<T> { // copied+simplified version of ReadableStream from @types/node/index.d.ts
-      readable: boolean; pipe: Function; unpipe: Function; wrap: Function;
-      read(size?: number): string | Uint8Array; setEncoding(encoding: string): this; pause(): this; resume(): this;
-      isPaused(): boolean; unshift(chunk: string | Uint8Array): void;
-    }
+interface WebStream<T extends Data> { // copied+simplified version of ReadableStream from lib.dom.d.ts
+  readonly locked: boolean; pipeThrough: Function; pipeTo: Function; tee: Function;
+  cancel(reason?: any): Promise<void>;
+  getReader(): ReadableStreamInternals.ReadableStreamDefaultReader<T>;
+}
 
-    type Stream<T extends Data> = WebStream<T> | NodeStream<T>;
+interface NodeStream<T extends Data> { // copied+simplified version of ReadableStream from @types/node/index.d.ts
```

Comment:
Probably we should keep
```suggestion
interface NodeStream<T extends Data> extends AsyncIterable<T> { // copied+simplified version of ReadableStream from @types/node/index.d.ts
```

And also, since we're removing `AsyncIterable` from `WebStream` (rightly so, as it isn't guaranteed to be there), we should update the examples, as per #1370.